node/pkg/p2p/cutover.go

```diff
@@ -0,0 +1,138 @@
+
+	if delay != time.Duration(0) {
+		// Wait for the cut over time and then panic so we restart with the new quic-v1. TODO: Can we just restart p2p??
+		go func() {
+			time.Sleep(delay)
+			logger.Info("time to cut over to new quic-v1", zap.String("cutOverTime", cutOverTimeStr), zap.String("component", "p2pco"))
+			panic("p2pco: time to cut over to new quic-v1")
+		}()
+	}
+
+	return newBootstrapPeers, newCcqBootstrapPeers, err
+}
+
+func checkForCutOverImpl(logger *zap.Logger, bootstrapPeers string, ccqBootstrapPeers string, components *Components, coTimeStr string, now time.Time) (newBootstrapPeers string, newCcqBootstrapPeers string, delay time.Duration, err error) {
+	newBootstrapPeers = bootstrapPeers
+	newCcqBootstrapPeers = ccqBootstrapPeers
+
+	if coTimeStr == "" {
+		return
+	}
+
+	bootstrapIsV1, err := validateQuic(bootstrapPeers)
+	if err != nil {
+		logger.Error(`bootstrap peers string is invalid:`, zap.String("bootstrapPeers", bootstrapPeers), zap.Error(err), zap.String("component", "p2pco"))
+		err = fmt.Errorf("unexpected format of bootstrap peers: %w", err)
+		return
+	}
+
+	if ccqBootstrapPeers != "" {
+		ccqBootstrapIsV1, ccqerr := validateQuic(ccqBootstrapPeers)
+		if ccqerr != nil {
+			logger.Error(`ccq bootstrap peers string is invalid:`, zap.String("ccqBootstrapPeers", ccqBootstrapPeers), zap.Error(ccqerr), zap.String("component", "p2pco"))
+			err = fmt.Errorf("unexpected format of ccq bootstrap peers: %w", ccqerr)
+			return
+		}
+
+		if bootstrapIsV1 != ccqBootstrapIsV1 {
+			logger.Error(`there is a mismatch between bootstrap peers and ccq bootstrap peers:`,
+				zap.Bool("bootstrapIsV1", bootstrapIsV1),
+				zap.Bool("ccqBootstrapIsV1", ccqBootstrapIsV1),
+				zap.String("bootstrapPeers", bootstrapPeers),
+				zap.String("ccqBootstrapPeers", ccqBootstrapPeers),
+				zap.Error(err), zap.String("component", "p2pco"),
+			)
+			err = fmt.Errorf("quic version mismatch between bootstrap peers and ccq bootstrap peers")
+			return
+		}
+	}
+
+	if bootstrapIsV1 {
+		for _, la := range components.ListeningAddressesPatterns {
+			if strings.Contains(la, "quic") && !strings.Contains(la, "quic-v1") {
+				err = fmt.Errorf("bootstrapPeers has been updated to quic-v1, but components.ListeningAddressesPatterns has not: %s", la)
+				return
+			}
+		}
+		logger.Info("bootstrap peers parameter is already using quic-v1, cut over is not necessary", zap.String("bootstrapPeers", bootstrapPeers), zap.String("component", "p2pco"))
+		delay = 0
+		return
+	}
+
+	cutOverTime, err := time.Parse(cutOverFmtStr, coTimeStr)
+	if err != nil {
+		logger.Error("failed to parse p2p cut over time", zap.String("cutOverTimeStr", coTimeStr), zap.Error(err), zap.String("component", "p2pco"))
+		err = fmt.Errorf("failed to parse cut over time: %w", err)
+		return
+	}
+
+	if cutOverTime.Before(now) {
+		// We should already be using the new quic-v1.
+		newBootstrapPeers = strings.ReplaceAll(bootstrapPeers, "quic", "quic-v1")
```

Comment:
should we add `/`s to all these? has a bit of a smell without them. what if someone's bootstrap domain has `quic` in it?

Reply:
i.e. bootstrap peers look like 

```
/dns4/wormhole-testnet-v2-bootstrap.certus.one/udp/8999/quic/p2p/12D3KooWAkB9ynDur1Jtoa97LBUp8RXdhzS5uHgAfdTquJbrbN7i,/dns4/t-guardian-01.nodes.stable.io/udp/8999/quic/p2p/12D3KooWCW3LGUtkCVkHZmVSZHzL3C4WRKWfqAiJPz1NR7dT9Bxh,/dns4/t-guardian-02.nodes.stable.io/udp/8999/quic/p2p/12D3KooWJXA6goBCiWM8ucjzc4jVUBSqL9Rri6UpjHbkMPErz5zK
```

so we wouldn't want to replace `/dns4/testnet-quic.certus.one/...` with `/dns4/testnet-quic-v1.certus.one/...`.

including the separators should be safer - `"/quic/","/quic-v1/"`